### PR TITLE
[Feature :#162727513] view and edit profile

### PR DESCRIPTION
**Pull request by habibaudu**
#### What does this PR do?
- view user profile and  edit user profile
 
#### Description of Task to be completed?
- [x] View user profile
- [x] Edit user profile
- [x] select picture to upload
- [x] view selected picture
- [x] upload picture to cloudinary
- [x] write tests

#### How should this be manually tested?
- On terminal run `npm run start:dev`
- goto `localhost:8080/profile`
- click update profile

#### What are the relevant pivotal tracker stories?
- [#162727513](https://www.pivotaltracker.com/story/show/162727513)

### CODE REVIEW

**Comment by rajeman** [comment]
@habibaudu Can you try increasing the coverage?
ok I will

File: package.json [hunk 1]
```diff
@@ -38,11 +38,14 @@
     "babel-jest": "^24.1.0",
     "babel-loader": "^8.0.5",
     "cloudinary": "^1.13.2",
+    "cloudinary-core": "^2.6.2",
+    "cloudinary-react": "^1.1.0",
```
Comment (andela-mabdussalam) [inline package.json]: Are we still using this?

File: src/app/profile/duck/reducers.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        ...state,
+        profileData,
+      };
+    }
+
+    case types.VIEW_USER_PROFILE: {
+      const { viewData } = action;
+      return {
+        ...state,
+        viewData,
```
Comment (andela-mabdussalam) [inline src/app/profile/duck/reducers.js]: How does viewData differ form profileData above?

File: src/app/profile/duck/operations.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+) => dispatch => {
+  const token = localStorage.getItem('token');
+  const defaultOptions = {
+    headers: {
+      Authorization: token,
+    },
+  };
+  return axios
+    .patch(
+      url2,
```
Comment (andela-mabdussalam) [inline src/app/profile/duck/operations.js]: Change the variable name `url2` to a more descriptive name.

File: src/app/profile/duck/operations.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+          status: constants.UPDATE_ERROR,
+        }),
+      );
+    });
+};
+
+export const viewProfile = () => dispatch => {
+  const token = localStorage.getItem('token');
+  const decoded = jwtDecode(token);
+  const url = `https://learnground-api-staging.herokuapp.com/api/v1/user/${
```
Comment (andela-mabdussalam) [inline src/app/profile/duck/operations.js]: Let us move this to an env variable.

File: src/app/profile/duck/operations.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      dispatch(viewUserProfile(data));
+    })
+    .catch(({ response }) => {
+      toast.error(response.data.error, {
+        hideProgressBar: true,
+      });
+    });
+};
+
+const url2 = 'https://learnground-api-staging.herokuapp.com/api/v1/profile';
```
Comment (andela-mabdussalam) [inline src/app/profile/duck/operations.js]: Same here, let us also use an env variable.

File: package.json (hunk 1 above)
Comment (habibaudu) [reply package.json]: No we are not... 
I will uninstall them

File: src/app/util/uploadToCloudinary.js [hunk 6]
```diff
@@ -26,16 +26,22 @@ export const saveImage = (page, imageUrl) => {
     });
 };
 
-const uploadImage = (page, imageData) => {
-  axios
-    .post(CLOUDINARY_URL, imageData, {
+const uploadImage = (page, imageUrl) => {
+  const formData = new FormData();
+  formData.append('tags', page);
```
Comment (andela-mabdussalam) [inline src/app/util/uploadToCloudinary.js]: I do not think we need this here again. Pull in develop and update.

File: src/testSetup.js [hunk 7]
```diff
@@ -3,3 +3,8 @@ import { configure } from 'enzyme';
 import Adapter from 'enzyme-adapter-react-16';
 
 configure({ adapter: new Adapter() });
+
+// global.FileReader = () => ({
```
Comment (andela-mabdussalam) [inline src/testSetup.js]: Commented code.

File: src/app/profile/duck/reducers.js (hunk 2 above)
Comment (habibaudu) [reply src/app/profile/duck/reducers.js]: ViewData is data gotten when making an API call to view user data while profile data is that sent when making a patch request to update user profile...

File: src/app/profile/duck/operations.js (hunk 3 above)
Comment (habibaudu) [reply src/app/profile/duck/operations.js]: Ok
I will

File: src/app/profile/duck/operations.js (hunk 4 above)
Comment (habibaudu) [reply src/app/profile/duck/operations.js]: ok

File: src/app/profile/duck/operations.js (hunk 5 above)
Comment (habibaudu) [reply src/app/profile/duck/operations.js]: ok

File: src/app/util/uploadToCloudinary.js (hunk 6 above)
Comment (habibaudu) [reply src/app/util/uploadToCloudinary.js]: ok

File: src/testSetup.js (hunk 7 above)
Comment (habibaudu) [reply src/testSetup.js]: Will take them out

**Comment by andela-mabdussalam** [comment]
@habibaudu This story includes notification changes which is out of the scope of this story. The coverage is impacted too.

**Comment by andela-mabdussalam** [comment]
@habibaudu See https://coveralls.io/builds/21880248/source?filename=src%2Fapp%2Fprofile%2FProfileComponent.jsx#L84 and https://coveralls.io/builds/21880248/source?filename=src%2Fapp%2Fprofile%2Fduck%2Foperations.js#L30

We could improve the test by ensuring those actions are tested.

**Comment by andela-mabdussalam** [comment]
@habibaudu I think you are editing my comments as opposed to writing yours.

**Comment by habibaudu** [comment]
> @habibaudu I think you are editing my comments as opposed to writing yours.

noted...
mistake

File: public/index.html [hunk 8]
```diff
@@ -41,12 +41,12 @@
         const instances = M.Sidenav.init(elems);
       });
     </script>
-    <script>
+    <!-- <script>
```
Comment (andela-mabdussalam) [inline public/index.html]: @habibaudu Since we no longer use this, let us comment it out.

Comment (habibaudu) [reply public/index.html]: ok

**Review by andela-mabdussalam — APPROVED** [review]

**Pull request merged**